src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Configuration/BaseProjectConfigurationDimensionProvider.cs

```diff
@@ -0,0 +1,120 @@
+        /// </summary>
+        /// <param name="projectXmlAccessor">Lock service for the project file.</param>
+        /// <param name="dimensionName">Name of the dimension.</param>
+        /// <param name="propertyName">Name of the project property containing the dimension values.</param>
+        public BaseProjectConfigurationDimensionProvider(IProjectXmlAccessor projectXmlAccessor, string dimensionName, string propertyName)
+        {
+            Requires.NotNull(projectXmlAccessor, nameof(projectXmlAccessor));
+            _projectXmlAccessor = projectXmlAccessor;
+
+            _dimensionName = dimensionName;
+            _propertyName = propertyName;
+        }
+
+        /// <summary>
+        /// Gets the property values for the dimension.
+        /// </summary>
+        /// <param name="unconfiguredProject">Unconfigured project.</param>
+        /// <returns>Collection of values for the dimension.</returns>
+        protected virtual async Task<ImmutableArray<string>> GetOrderedPropertyValuesAsync(UnconfiguredProject unconfiguredProject)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+
+            string propertyValue = await GetPropertyValue(unconfiguredProject).ConfigureAwait(true);
+            if (propertyValue == null || string.IsNullOrEmpty(propertyValue))
+            {
+                return ImmutableArray<string>.Empty;
+            }
+            else
+            {
+                return MsBuildUtilities.GetPropertyValues(propertyValue);
+            }
+        }
+
+        /// <summary>
+        /// Gets the defaults values for project configuration dimensions for the given unconfigured project.
+        /// </summary>
+        /// <param name="unconfiguredProject">Unconfigured project.</param>
+        /// <returns>Collection of key/value pairs for the defaults values for the configuration dimensions of this provider for given project.</returns>
+        public virtual async Task<IEnumerable<KeyValuePair<string, string>>> GetDefaultValuesForDimensionsAsync(UnconfiguredProject unconfiguredProject)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+
+            var values = await GetOrderedPropertyValuesAsync(unconfiguredProject).ConfigureAwait(false);
+            if (values.IsEmpty)
+            {
+                return ImmutableArray<KeyValuePair<string, string>>.Empty;
+            }
+            else
+            {
+                // First value is the default one.
+                var defaultValues = ImmutableArray.CreateBuilder<KeyValuePair<string, string>>();
+                defaultValues.Add(new KeyValuePair<string, string>(_dimensionName, values.First()));
+                return defaultValues.ToImmutable();
+            }
+        }
+
+        /// <summary>
+        /// Gets the project configuration dimensions for the given unconfigured project.
```

Comment:
The description of this method is confusing. This method returns the value(s) for a given dimension in a project. The existing description, `Gets the project configuration dimensions...`, and also the method name, seems to suggest that we are providing the different dimensions itself, For eg: `Configurations`, `Platforms`, rather than the value for those dimensions.